### PR TITLE
UI styling and landing page improvement(with future ideas)

### LINKED ISSUE #45: Making the website responsive and interactive
### Is this a unique feature?

- [X] I have checked "open" AND "closed" issues and this is not a duplicate

### Is your feature request related to a problem/unavailable functionality? Please describe.

The UI/UX is a bit bland, I Saw an issue raised with adding flexbox and more colour gradients. We could add some features to make it more interactive.
Also there is not much info about the repo on main page.

### Proposed Solution

-Added an assets folder with starImage.svg and forkImage.svg to make the UI more pleasent.
Added Hover effect on Repo cards so that they scale-up on hovering.
Added the prominent language used in the repo so users find it easier to select the repositries they are interested in.

### Screenshots

Screenshot_20241003_104258_GitHub

### Do you want to work on this issue?

Yes

### If "yes" to above, please explain how you would technically implement this (issue will not be assigned if this is skipped)

Already made the changes and Added a PR.

### PR DESCRIPTION
**Pull request by AdarshRaj241408**
## Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

- Added an assets folder with starImage.svg and forkImage.svg to make the UI more pleasent.
- Added Hover effect on Repo cards so that they scale-up on hovering.
- Added the prominent language used in the repo so users find it easier to select the repositries they are interested in.

### Future Changes
- Adding Filters based on Recent Activity and language used.
- Utilising this website for more than just Repo searching as there is a lot of empty space on the right, maybe transition towards backend and show your github stats on the right side, utilising empty space efficiently and giving more backbone to the website.

Fixes #45 

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

## Screenshots (if applicable):
image

### CODE REVIEW

**Comment by Bashamega** [comment]
Please open an issue first for something this big so we can discuss

**Comment by Bashamega** [comment]
@AdarshRaj241408 This is the pr for #45 correct?

**Comment by AdarshRaj241408** [comment]
#45

**Comment by AdarshRaj241408** [comment]
Yes

File: package-lock.json
Comment (Bashamega) [inline package-lock.json:1]: please do not update this file

File: src/components/Repo.tsx [hunk 1]
```diff
@@ -12,12 +15,32 @@ const Repo: React.FC<RepoProps> = ({ data, key }) => {
...
     >
-      <p className="text-xl font-semibold text-gray-200">{data.name}</p>
-      <p className="text-gray-400">Stars: {data.stars}</p>
-      <p className="text-gray-400">Forks: {data.forks}</p>
+      <div className="flex items-center">
+        <p className="text-xl font-semibold text-gray-200">{data.name}</p>
+
+        {/* Language Badge - Only show if language is specified */}
+        {data.language ? (
+          <span className="ml-2 inline-block bg-blue-500 text-white text-xs font-medium px-2.5 py-0.5 rounded">
```
Comment (Bashamega) [inline src/components/Repo.tsx:26]: Use something like [github-colors](https://www.npmjs.com/package/github-colors) for custom bg colors.

**Review by Bashamega — CHANGES_REQUESTED** [review]
Make the bg responsive
image

File: src/components/Repo.tsx (hunk 1 above)
Comment (AdarshRaj241408) [reply src/components/Repo.tsx:26]: Done!

**Comment by AdarshRaj241408** [comment]
Made the requested changes. The colours for the languages are custom and can be changed in the new src/color.ts file,
the backgroun has been made responsive as hover adds a dark blue tint and the repo-car expans.
Screenshot from 2024-10-03 12-19-01

**Review by Bashamega — APPROVED** [review]
Hello:)
Great work, i just modified some things.

**Pull request merged**